packages/core/src/StateNode.ts

```diff
@@ -178,17 +176,17 @@ export class StateNode<
         ? 'history'
         : 'atomic');
 
+    this.order = this.machine.idMap.size;
```

Comment:
quite an improvement over `dfs` algorithm 👍

Reply:
Yeah, that was a bit ugly 🙈